app/models/tree_resolver.rb

```diff
@@ -25,53 +22,78 @@ def enqueue_tree_resolution
   end
 
   def load_dependencies_tree
-    tree_data = Rails.cache.fetch cache_key, :expires_in => 1.day, race_condition_ttl: 2.minutes do
-      generate_dependency_tree
-    end
+    tree_data = Rails.cache.fetch(
+      cache_key,
+      expires_in: 1.day,
+      race_condition_ttl: 2.minutes,
+      &method(:generate_dependency_tree)
+    )
     @project_names = tree_data[:project_names]
     @license_names = tree_data[:license_names]
     @tree = tree_data[:tree]
   end
 
+  def project_names
+    @project_names.to_a
+  end
+
+  def license_names
+    @license_names.to_a
+  end
+
   private
 
-  def generate_dependency_tree
+  def generate_dependency_tree(_key)
```

Comment:
_key ?

Reply:
Rails cache `fetch` passes the key to the block

Reply:
Ahhh, gotcha